public/app/features/scopes/selector/ScopesSelectorService.ts

```diff
@@ -147,50 +131,44 @@ export class ScopesSelectorService extends ScopesServiceBase<ScopesSelectorServi
     }
   };
 
-  public changeScopes = (scopeNames: string[]) =>
-    this.setNewScopes(scopeNames.map((scopeName) => ({ scopeName, path: [] })));
+  changeScopes = (scopeNames: string[]) => this.setNewScopes(scopeNames.map((scopeName) => ({ scopeName, path: [] })));
 
-  public setNewScopes = async (treeScopes = this.state.treeScopes) => {
+  private setNewScopes = async (treeScopes = this.state.treeScopes) => {
     if (isEqual(treeScopes, this.getTreeScopesFromSelectedScopes(this.state.selectedScopes))) {
       return;
     }
 
     let selectedScopes = treeScopes.map(({ scopeName, path }) => ({
-      scope: this.getBasicScope(scopeName),
+      scope: getEmptyScopeObject(scopeName),
       path,
     }));
-    this.updateState({ selectedScopes, treeScopes });
-    ScopesService.instance?.setLoading(true);
-    ScopesDashboardsService.instance?.fetchDashboards(selectedScopes.map(({ scope }) => scope.metadata.name));
-
-    selectedScopes = await this.fetchScopesApi(treeScopes);
-    this.updateState({ selectedScopes });
-    ScopesService.instance?.setScopes(selectedScopes.map(({ scope }) => scope));
-    ScopesService.instance?.setLoading(false);
+    this.updateState({ selectedScopes, treeScopes, loading: true });
+    this.dashboardsService.fetchDashboards(selectedScopes.map(({ scope }) => scope.metadata.name));
+
+    selectedScopes = await this.apiClient.fetchMultipleScopes(treeScopes);
+    this.updateState({ selectedScopes, loading: false });
   };
 
   public removeAllScopes = () => this.setNewScopes([]);
 
   public open = async () => {
-    if (!ScopesService.instance?.state.readOnly) {
-      if (Object.keys(this.state.nodes[''].nodes).length === 0) {
-        await this.updateNode([''], true, '');
-      }
+    if (Object.keys(this.state.nodes[''].nodes).length === 0) {
```

Comment:
just removed the readOnly condition to remove the ScopeService dependency and moved that check to the react component which already needed those services anyway.